### DISCUSSION TITLE
v2.0.0-beta.27

### DISCUSSION THREAD

**Discussion by 0xJacky** [0xJacky/nginx-ui#448] [Announcements]
# Notice
We're excited to announce that we've been hard at work developing the next generation of the Nginx UI (v2). To keep up-to-date with our progress, please visit https://github.com/users/0xJacky/projects/1.

Please feel free to send any feedbacks, suggestions or bug reports in Github issues.

# Change vs v2.0.0-beta.26
1. Remove trailing `.conf` in default nginx stream block, thanks to the contributions of @PeterWang-dev. #444 #445 
2. Enhance(upgrader): gracefully replace the old exe with the new exe.
3. Add batch upgrade remote nginx-ui.
4. Hide enviroments management page and upgrade page when connecting to remote node.
5. Fix the issue of saving settings #446.
6. Add 2FA authorization for login #344, #419.
7. Add 2FA authorization for Web Terminal #394 .

# Breaking Changes:
1. About PR #445, if you are using nginx-ui docker container and encounter this issue, you have to remove the trailing `.conf` manually. Diff: https://github.com/0xJacky/nginx-ui/pull/445/files
2. The `nginx_log` section in the configuration file has been renamed to `nginx`.
`NginxConfigDir` and `NginxPIDPath` in the `server` section have been renamed to `ConfigDir` and `PIDPath` respectively, and both have been moved into the `nginx` section.
3. The `ReloadCmd` and `RestartCmd` have been added in the `nginx` section to configure custom commands for reloading/restarting Nginx.
8. In versions above v2.0.0-beta.23 of Nginx UI Docker containers, `/var/logs/nginx/access.log` and `/var/logs/nginx/error.log` are no longer soft links to `/dev/stdout` and `/dev/stderr`, but regular files. If you need this functionality, please add `access_log /dev/stdout` and `error_log /dev/stderr` in your `nginx.conf`. For users upgrading from older versions who do not need this feature, check if your `nginx.conf` contains `access_log /var/logs/nginx/access.local.log` and `error_log /var/logs/nginx/error.local.log`. If so, please manually remove them to avoid duplicate log entries. This change allows you to map `/var/logs/nginx` to a directory on the host machine to achieve persistent storage of log files.

# What's new:
1. We're proud to have registered the [nginxui.com](https://nginxui.com) domain 🎉.
10. We have launched a new documentation site, please check [nginxui.com](https://nginxui.com). 
11. Node management functionality has been introduced (#70).
12. Duplicate site function now can duplicate config to remote server.
13. French translations have been added, thanks to the contributions of @skrilax91.
14. Added border-radius to Ace Editor and terminal, we hope you will like this change :).
15.  You can now deploy site configuration to a remote server.
16. Spanish translations have been added, thanks to the contributions of @kcholoren.
17. Russian translations have been added, thanks to the contributions of @iAHTOH.
18. The dashboard now includes a system monitor for nodes.
19. Use jsoniter for improving gin performance.
20. Added custom reload/restart command for nginx. #140
21. Refacted translation module #135 
22. Added http host config #157 by @quanbisen 
23. Add setting for test nginx config command #191 
24. Added casdoor sso support, thanks to the contributions of @Jraaay 
25. Refactor the project directory structure.
26. Use tailwind css.
27. Use eslint to check the code style of frontend.
28. Use vue-tsc to check the types for frontend.
29.  Introducing a new management page for certificates, user can now view the latest renew logs of the certain certificate.
30. Add manually renew button in certificate modify page for managed certificate (auto cert).
31. Add notifications for renew certificates. #192 
32. Automatically detect the path of accress logs and error logs.
33. Introducing the feature of issuing wildcard certificates on the Manage Certificate page. #190 
34. Add some Nginx Config Templates #233, thanks to the contributions of @sanvu88 . 
35. Support different key types for obtaining certificate, including RSA2048, RAS3072, RSA4096, EC256 and EC384 #264.
36. Introducing logroate cron task for docker users #255.
37. A system default acme user will be registered when nginx ui start. https://github.com/0xJacky/nginx-ui/issues/237
38. Add acme user management page. https://github.com/0xJacky/nginx-ui/issues/319
39. Persist the acme user to the database.
40. Add a setting of renewal interval of certificates (7-21d). https://github.com/0xJacky/nginx-ui/issues/343
41. Add a option to select a acme user when obtaining a certificate. https://github.com/0xJacky/nginx-ui/issues/319 https://github.com/0xJacky/nginx-ui/issues/16
42. SSL management support different types of certificates of a same doamin name. https://github.com/0xJacky/nginx-ui/issues/309, https://github.com/0xJacky/nginx-ui/issues/288
43. Certificates obtained within 21 days are renewed by using the renewal api. For compatibility, the challenge resolver is still setup when renewing. https://github.com/0xJacky/nginx-ui/issues/319
44. Add trashs for soft deleted items in CURD pages.
45. Use s6-overlay to manage the services in docker #308 .
46. Users can use settings or environment variable to skip installation #357 .
47. Users can use environment variable to predefine a admin user  (only available in skip installation mode) #214 .
48. Users can customize local environment name #313 .
49. Users can use settings file to predefine nodes #169 .
50. Hot reload cluster node settings from settings file #169.
51. Introducing synchronize certificates to remote server #293, #363, #411.
52. Enabled search box for Ace code edior #412.
54. Add a white list of IPs, only those in list can access the Nginx UI. 
55. Add login failed ban IP list. 
56. Add 2FA authorization for login #344, #419.
57. Add 2FA authorization for Web Terminal #394 .

# Fixes:
1. We've upgraded gonginx to address the configuration parsing issue #73.
9. Fixed an issue where it was not possible to update settings value to an empty string.
10. Resolved a location saving issue on the domain addition page #116.
11. Site enabling/disabling is now aborted only when the log level is higher than warning.
12. Fixed environment indicator display overlay and data obtaining issues #152 by @quanbisen 
13. Fixed env indicator background in dark mode.
14. Terminal process not killed #181 
15. Added `gpt-4-1106-preview` model in ChatGPT settings.
16. Fixed change language cause unexpected redirect #189 
17. Fixed english grammar error in button text in sites management page. #200 
18. Fixed logs mnual link error. #201 
19. Revert enable site if you trying to enable a incorrect config of the site. #199 
20. Upgrade ant design vue verison from v3 to v4, thanks @funcfang
21. Fixed FooterToolBar content was covered by Monica, a Chrome Extension. #195 
22. Delete server confirm modal not destroy #206 
23. clientMaxBodySize name is wrong. #216 
24. Fix error maybe occurs when enable TLS. #230 
25. Introducing streams management pages. #166 
26. Introducing upstream editor. #138 
27. Introducing upstream health check (supports TCP and Unix Socket).
28. Fix a bug that may casue nginx-ui taking 100% CPU if the log file is not a regular file.

# Enhances:
1. The recovery function now can print call stack information.
7. Added file info to config editor.
8. Improved zh_TW locale. #148 
9. Enhance error handler for execShell.
10. Enhance nginx control and status acquisition. #222 
11. Remove the vaildated rules for DNS Credentials

# Docs:
1. Traditional Chinese documentation has been added, thanks to @Hintay.
6. We've added documentation for the syntax of the configuration template.
7. We have updated the documentation for the nginx proxy example.
12. Updated deprecated TLS directives #183 

# Others:
1. Core upgrades will be performed as a dry run in demo mode.
4. Updated zh_CN translations.
6. Changed Token to NodeSecret #152 
9. Updated configurations of DNS challenges provider.

<hr /><em>This discussion was created from the release <a href='https://github.com/0xJacky/nginx-ui/releases/tag/v2.0.0-beta.27'>v2.0.0-beta.27</a>.</em>

**Comment by leic4u** [comment]
Hey jacky, when will the beta version be upgraded to the official version?

**Reply by 0xJacky** [reply]
Today.

**Reply by leic4u** [reply]
> Today.

![image](https://github.com/user-attachments/assets/2133df0d-aff8-43b3-bcef-e785eed50938)

**Reply by 0xJacky** [reply]
Already released as stable version.
